### PR TITLE
Added disableMainThreadChecker to TargetScheme and Run/Test Actions

### PR DESCRIPTION
**Pull request by wag-miles**
Added capability to disable the main thread checker. Added values to test cases where appropriate.

### CODE REVIEW

File: Sources/ProjectSpec/Scheme.swift [hunk 1]
```diff
@@ -294,6 +306,7 @@ extension Scheme.Test: JSONEncodable {
     public func toJSONValue() -> Any {
         return [
             "gatherCoverageData": gatherCoverageData,
+            "disableMainThreadChecker": disableMainThreadChecker,
```
Comment (yonaskolb) [inline Sources/ProjectSpec/Scheme.swift]: It looks like `gatherCoverageData` accidentally missed out on this before, so it was easy to do, but these values should only be written if they differ from the default

Comment (yonaskolb) [reply Sources/ProjectSpec/Scheme.swift]: Also is `Scheme.Run` writing this property?

File: Docs/ProjectSpec.md [hunk 2]
```diff
@@ -686,6 +687,7 @@ A multiline script can be written using the various YAML multiline methods, for
 ### Test Action
 
 - [ ] **gatherCoverageData**: **Bool** - a boolean that indicates if this scheme should gather coverage data. This defaults to false
+- [ ] **disableMainThreadChecker**: **Bool** - a boolean that indicates if this scheme should disable disable the Main Thread Checker. This defaults to false
```
Comment (yonaskolb) [inline Docs/ProjectSpec.md]: From the code it looks like this is also for the `Run Action` yes?

**Comment by wag-miles** [comment]
I believe I've covered your suggested changes. Thanks!

**Comment by yonaskolb** [comment]
Thanks for the changes @wag-miles. One more thing: Could you edit a scheme in here to set this property https://github.com/yonaskolb/XcodeGen/blob/master/Tests/Fixtures/TestProject/project.yml and then re-run the tests and commit the project diff.
This will serve as an integration test

**Comment by wag-miles** [comment]
> Thanks for the changes @wag-miles. One more thing: Could you edit a scheme in here to set this property https://github.com/yonaskolb/XcodeGen/blob/master/Tests/Fixtures/TestProject/project.yml and then re-run the tests and commit the project diff.
> This will serve as an integration test

Done. Thanks!

**Comment by bartzet** [comment]
Hi @yonaskolb 👋

Do you have an idea when this PR will be merged? We need this flag to be available as after migrating to Xcode 11 we have noticed some tests failing and to fix them we need to disableMainThreadChecker when we set up the project with XcodeGen. 

Please provide more details.

**Comment by yonaskolb** [comment]
@bartzet this is now released in 2.7.0 👍

**Comment by bartzet** [comment]
Amazing. Thank you for acting that quickly @yonaskolb!

**Pull request merged**